Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -228,9 +236,23 @@ private CachedCoinView(ILoggerFactory loggerFactory, StakeChainStore stakeChainS
         /// <summary>
         /// Finds all changed records in the cache and persists them to the underlaying coinview.
         /// </summary>
-        public async Task FlushAsync()
+        /// <param name="force"><c>true</c> to enforce flush, <c>false</c> to flush only if <see cref="lastCacheFlushTime"/> is older than <see cref="CacheFlushTimeIntervalSeconds"/>.</param>
+        /// <remarks>
+        /// WARNING: This method can only be run from <see cref="ConsensusLoop.Execute(System.Threading.CancellationToken)"/> thread context
+        /// or when consensus loop is stopped. Otherwise, there is a risk of race condition when the consensus loop accepts new block.
+        /// </remarks>
+        public async Task FlushAsync(bool force = true)
         {
-            this.logger.LogTrace("()");
+            this.logger.LogTrace("({0}:{1})", nameof(force), force);
+
+            DateTime now = this.dateTimeProvider.GetUtcNow();
+            if (!force && ((now - this.lastCacheFlushTime).TotalSeconds < CacheFlushTimeIntervalSeconds))
+            {
+                this.logger.LogTrace("(-)[NOT_NOW]");
+                return;
+            }
+
+            this.lastCacheFlushTime = now;
```

Comment:
I'd move this to the bottom of the method.

Reply:
It should not matter. At the end of the method, I would tend to use GetUtcNow again, that is maybe what you actually suggest, isn't it?

Reply:
OK, I will move it. The problem is that I'm unsure about the second return there is for null inner tip. I am not familiar with  the code enough to decide when this actually happens and if we should reset lastCFT in that branch as well. Hopefully not, so I will just set it at the end of the method.

Reply:
yeah just so that you get the latest time.